util/util.go

```diff
@@ -437,6 +437,20 @@ func HasOwner(refList []metav1.OwnerReference, apiVersion string, kinds []string
 	return false
 }
 
+// HasOwnerWithSameVersion checks if any of the references in the passed list match the given group from apiVersion and one of the given kinds.
+func HasOwnerWithSameVersion(refList []metav1.OwnerReference, apiVersion string, kinds []string) bool {
```

Comment:
I know it's relatively trivial but let's ~ copy&paste the unit tests of HasOwner